kafka/util.py

```diff
@@ -151,3 +151,18 @@ def stop(self):
 
     def __del__(self):
         self.stop()
+
+class EventRegistrar(object):
+    """
+    Handles registration of callable event handlers which are
+    executed in sequence when events are emitted.
+    """
+    def __init__(self):
+        self.handlers = collections.defaultdict(list)
+
+    def register(self, event, handler):
+        self.handlers[event].append(handler)
+
+    def emit(self, event, *args, **kwargs):
+        for handler in self.handlers[event]:
+            handler(*args, **kwargs)
```

Comment:
we should wrap in try / except to prevent an errant handler from crashing the producer thread